package-res/resources/web/pentaho/type/_type.js

```diff
@@ -0,0 +1,1362 @@
+      // "" -> null conversion
+
+      _category: null,
+
+      /**
+       * Gets or sets the category associated with this type.
+       *
+       * The category is used primarily to group similar types (or instances of) in a user interface.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
+       * thus ignoring any inherited value.
+       *
+       * @type {?nonEmptyString}
+       * @see pentaho.type.Type#isBrowsable
+       * @see pentaho.type.Type#ordinal
+       */
+      get category() {
+        return this._category;
+      },
+
+      set category(value) {
+        if(value === undefined) {
+          this._resetCategory();
+        } else {
+          this._category = nonEmptyString(value);
+        }
+      },
+
+      _resetCategory: function() {
+        if(this !== _type) {
+          delete this._category;
+        }
+      },
+      //endregion
+
+      //region helpUrl property
+
+      // -> nonEmptyString, Optional, Inherited, Configurable, Localized?
+      // "" -> null conversion
+
+      _helpUrl: null,
+
+      /**
+       * Gets or sets an URL pointing to documentation associated with this type.
+       *
+       * Attempting to set to a non-string value type implicitly
+       * converts the value to a string before assignment.
+       *
+       * Setting to `undefined` causes this attribute to use the inherited value,
+       * except for the root type, `Instance.type` (which has no ancestor), where the attribute is `null`.
+       *
+       * Setting to `null` or to an empty string clears the attribute and sets it to `null`,
+       * ignoring any inherited value.
+       *
+       * @type {?nonEmptyString}
+       */
+      get helpUrl() {
+        return this._helpUrl;
+      },
+
+      set helpUrl(value) {
+        if(value === undefined) {
+          this._resetHelpUrl();
+        } else {
+          this._helpUrl = nonEmptyString(value);
+        }
+      },
+
+      _resetHelpUrl: function() {
+        if(this !== _type) {
+          delete this._helpUrl;
+        }
+      },
+      //endregion
+
+      //region isBrowsable property
+      // @type boolean
+      // -> boolean, Optional(true), Inherited, Configurable
+      // undefined or null -> resets
+
+      _isBrowsable: true,
+
+      /**
+       * Gets or sets the `isBrowsable` attribute of this type.
+       *
+       * Browsable types are exposed to the end user.
+       * Set this attribute to `false` to prevent exposing the type in a user interface.
+       *
+       * Setting to a {@link Nully} value causes this attribute to use the inherited value,
```

Comment:
Suggest:

Setting this attribute to a {@link Nully} value causes this attribute to use the inherited value,